ix-dev/community/windmill/templates/docker-compose.yaml

```diff
@@ -0,0 +1,176 @@
+  "database": values.consts.db_name,
+  "volume": values.storage.postgres_data
+} %}
+{% set postgres = tpl.deps.postgres(
+  values.consts.postgres_container_name,
+  values.windmill.postgres_image_selector,
+  pg_config, perm_container
+) %}
+
+{% set proto = "https" if values.network.certificate_id else "http" %}
+{% set database_url = "postgres://" + values.consts.db_user + ":" + values.windmill.db_password + "@" + values.consts.postgres_container_name + "/" + values.consts.db_name + "?sslmode=disable" %}
+
+{# Windmill Server Container #}
+{% set c_server = tpl.add_container(values.consts.windmill_server_container_name, "image") %}
+{% do c_server.set_user(values.run_as.user, values.run_as.group) %}
+{% do c_server.healthcheck.set_test("wget", {"port": 8000, "path": "/api/health", "scheme": "http"}) %}
+
+{% do c_server.environment.add_env("DATABASE_URL", database_url) %}
+{% do c_server.environment.add_env("MODE", "server") %}
+{# TZ is already defined automatically from the library #}
+{% do c_server.environment.add_user_envs(values.windmill.additional_envs) %}
+
+{% do c_server.depends.add_dependency(values.consts.postgres_container_name, "service_healthy") %}
+
+{% do c_server.add_storage("/tmp/windmill/logs", values.storage.worker_logs) %}
+{% do perm_container.add_or_skip_action("worker_logs", values.storage.worker_logs, {"uid": values.run_as.user, "gid": values.run_as.group, "mode": "check"}) %}
+
+{% do c_server.expose.add_port(8000) %}
+{% do c_server.expose.add_port(2525) %}
+
+{# Windmill Worker Container #}
+{% set c_worker = tpl.add_container(values.consts.windmill_worker_container_name, "image") %}
+{% do c_worker.set_user(values.run_as.user, values.run_as.group) %}
+{% do c_worker.healthcheck.set_test("wget", {"port": 8000, "path": "/api/health", "scheme": "http"}) %}
+
+{% do c_worker.environment.add_env("DATABASE_URL", database_url) %}
+{% do c_worker.environment.add_env("MODE", "worker") %}
+{% do c_worker.environment.add_env("WORKER_GROUP", "default") %}
+{# TZ is already defined automatically from the library #}
+{% do c_worker.environment.add_user_envs(values.windmill.additional_envs) %}
+
+{% do c_worker.depends.add_dependency(values.consts.postgres_container_name, "service_healthy") %}
+
+{% do c_worker.add_storage("/tmp/windmill/cache", values.storage.worker_cache) %}
+{% do perm_container.add_or_skip_action("worker_cache", values.storage.worker_cache, {"uid": values.run_as.user, "gid": values.run_as.group, "mode": "check"}) %}
+
+{% do c_worker.add_storage("/tmp/windmill/logs", values.storage.worker_logs) %}
+
+{% if values.worker.mount_docker_socket %}
+{% do c_worker.add_storage("/var/run/docker.sock", {"type": "host_path", "host_path_config": {"path": "/var/run/docker.sock"}}) %}
+{% endif %}
+
+{# Commented out due to missing resources attribute
+{% do c_worker.resources.set_limits({"cpu": values.resources.limits.worker_cpus, "memory": values.resources.limits.worker_memory}) %}
```

Comment:
You should leave resources the same as all the other apps. And it will be picked up automatically.
Please don't split it for each container.

Later on we might add support for per container resources on all apps.

Reply:
can you elaborate? workers might consume significantly more resources and i found it sensible to account for that

Reply:
Resources will apply to all containers,
ie if you assign 2 cpus, each container will have 2 cpus.
Given that its not "reserving" those 2 cpus, its fine.

So in this case, user can assign 10 or more if it has intensive workload.

Later on if/when support for per container resources is added, it will be done in a way that will migrate/apply seamlessly.
If you do custom things here, it will only make migrations for me or whoever writes those hard(er)